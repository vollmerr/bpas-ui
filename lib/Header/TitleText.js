var _templateObject = _taggedTemplateLiteral(['\n  color: ', ';\n  font-size: ', 'px;\n  font-weight: 400;\n  padding: ', 'px;\n  margin: 0;\n  line-height: inherit;\n'], ['\n  color: ', ';\n  font-size: ', 'px;\n  font-weight: 400;\n  padding: ', 'px;\n  margin: 0;\n  line-height: inherit;\n']);

function _taggedTemplateLiteral(strings, raw) { return Object.freeze(Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } })); }

import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import theme from '../../util/theme';

var Text = styled.h1(_templateObject, theme.color.grey.white, theme.font.lg, theme.size.xs);

/** Renders the title bars title  */
function TitleText(_ref) {
  var title = _ref.title;

  return React.createElement(
    Text,
    null,
    title
  );
}

TitleText.propTypes = {
  title: PropTypes.string.isRequired
};

export default TitleText;