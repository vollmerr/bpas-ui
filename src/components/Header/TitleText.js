import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import logo from '../../../public/img/Ca-Gov-Logo-Gold.svg';
import theme from '../../theme';

const Wrapper = styled.div`
  display: flex;
`;

const Image = styled.img`
  height: ${theme.size.titleBar}px;
  padding: ${theme.size.xs}px;
`;

const Text = styled.h1`
  color: ${theme.color.primaryInverse};
  font-size: ${theme.font.lg}px;
  padding: ${theme.size.xs}px;
  margin: 0;
  line-height: inherit;
`;

/** Renders the headers title (and logo)  */
function TitleText({
  title,
  isMobile = false,
}) {
  return (
    <Wrapper>
      <a href={'http://www.ca.gov'}>
        <Image src={logo} alt={'CA Gov Logo'} />
      </a>
      {!isMobile && <Text>{title}</Text>}
    </Wrapper>
  );
}

TitleText.propTypes = {
  title: PropTypes.string.isRequired,
  isMobile: PropTypes.bool,
};

export default TitleText;
