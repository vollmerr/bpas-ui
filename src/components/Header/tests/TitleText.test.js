import React from 'react';
import { shallow } from 'enzyme';
import TitleText from '../TitleText';

const testTitle = 'test title';

describe('<TitleText />', () => {
  it('renders correctly', () => {
    const tree = shallow(
      <TitleText title={testTitle} />
    );
    expect(tree).toMatchSnapshot();
  });
});
