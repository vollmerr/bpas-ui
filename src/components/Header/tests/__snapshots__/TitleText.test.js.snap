// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TitleText /> renders correctly 1`] = `
<styled.h1>
  test title
</styled.h1>
`;
